resource-managers/yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnClientSchedulerBackend.scala

```diff
@@ -122,6 +122,13 @@ private[spark] class YarnClientSchedulerBackend(
         }
         allowInterrupt = false
         sc.stop()
+        state match {
+          case FinalApplicationStatus.FAILED | FinalApplicationStatus.KILLED =>
+            logWarning(s"ApplicationMaster finished with status ${state}, " +
+              s"SparkContext should exit with code 1.")
+            System.exit(1)
```

Comment:
If it comes from cli or api call, it may not be good to exit directly

Reply:
> If it comes from cli or api call, it may not be good to exit directly

You mean there may have situation, user's code call kill and just let sc stop?

Hmm， it's  a confusing problem...

Reply:
yeah ok so we could be in the situation where someone is using Spark as part of another application (e.g. some analytics). I think a good compromise for now could be adding another flag for "exit on driver error" or just switching this to a logError and dropping `system.exit`. What do folks think?

Reply:
> yeah ok so we could be in the situation where someone is using Spark as part of another application (e.g. some analytics). I think a good compromise for now could be adding another flag for "exit on driver error" or just switching this to a logError and dropping `system.exit`. What do folks think?

You mean change the configuration to `exit on driver error` and default is true, when false, if application master kill/failed, driver exit too?